html/js/product-search.js

```diff
@@ -157,52 +162,62 @@ function display_products(target, product_groups, use_user_product_preferences_f
 		$.each( product_group, function(key, product) {
 		
 			var product_html = "";
-			
+
 			// Show the green / grey / colors for matching products only if we are using the user preferences
-			if (use_user_product_preferences_for_ranking) {
-				product_html += '<li><a href="' + product.url + '" class="list_product_a list_product_a_match_' + product.match_status + '">';
-			}
-			else {
-				product_html += '<li><a href="' + product.url + '" class="list_product_a">';
+			let css_classes = 'list_product_a';
+			if (user_prefs.use_ranking) {
+				css_classes += ' list_product_a_match_' + product.match_status;
 			}
+			product_html += `<li><a href=${product.url}" class="${css_classes}">`;
 			product_html += '<div class="list_product_img_div">';
-			
-			if (product.image_front_thumb_url) {
-				product_html += '<img src="' + product.image_front_thumb_url + '" class="list_product_img">';
-			}
-			else {
-				product_html += '<img src="/images/icons/product-silhouette-transparent.svg" class="list_product_img">';
-			}
-			
+
+			const img_src =
+				product.image_front_thumb_url ||
+				"/images/icons/product-silhouette-transparent.svg"
+			;
+			product_html += `<img src="${img_src}" class="list_product_img">`;
+
 			product_html += "</div>";
-			
+
 			if (product.product_display_name) {
 				product_html += '<div class="list_product_name">' + product.product_display_name + "</div>";
 			}
 			else {
 				product_html += '<div class="list_product_name">' + product.code + "</div>";
 			}
-									
+
 			$.each(product.match_attributes.mandatory.concat(product.match_attributes.very_important, product.match_attributes.important), function (key, attribute) {
-				
+
 				if (attribute.icon_url) {
 					var title = attribute.title;
-				
+
 					if (attribute.description_short) {
 						title += ' - ' + attribute.description_short;
 					}
 
 					if (attribute.missing) {
 						title += " - " + attribute.missing;
-					}		
-					
+					}
+
 					product_html += '<img class="list_product_icons" src="' + attribute.icon_url + '" title="' + title + '">';
 				}
 			});
-			
-			product_html += "</a></li>";
-				
-			products_html.push(product_html);		
+			// add some specific fields
+			if (user_prefs.display.display_barcode) {
+				product_html += `<span class="list_display_barcode">${product.code}</span>`;
+			}
+			product_html += "</a>";
+			if (user_prefs.display.edit_link) {
+				const edit_url = product_edit_url(product);
+				product_html += `
+				<a class="list_edit_link" alt="Edit ${product.product_display_name}" href="${edit_url}">
```

Comment:
Can you add a title so that it's shown on hover?

You can get the translation like this:

+ '" title="' + lang()["edit_product_page"]